src/alloc/isolated_alloc.rs

```diff
@@ -255,18 +279,19 @@ impl IsolatedAlloc {
     /// SAFETY: Same as `dealloc()` with the added requirement that `layout`
     /// must ask for a size larger than the host pagesize.
     unsafe fn dealloc_huge(&mut self, ptr: *mut u8, layout: Layout) {
-        let layout = IsolatedAlloc::huge_normalized_layout(layout, self.page_size);
+        let size = self.huge_normalized_layout(layout, self.page_size);
         // Find the pointer matching in address with the one we got
         let idx = self
             .huge_ptrs
             .iter()
             .position(|pg| ptr.addr() == pg.0.addr())
             .expect("Freeing unallocated pages");
         // And kick it from the list
-        self.huge_ptrs.remove(idx);
-        // SAFETY: Caller ensures validity of the layout
+        let (un_offset_ptr, _) = self.huge_ptrs.remove(idx);
+        // SAFETY: huge_normalized_layout ensures the validity of the size, and
+        // we know the pointer was the original page-aligned one
         unsafe {
-            alloc::dealloc(ptr, layout);
+            assert_eq!(libc::munmap(un_offset_ptr.cast(), size), 0);
```

Comment:
```suggestion
            let ret = libc::munmap(un_offset_ptr.cast(), size);
            assert_eq!(ret, 0);
```
I prefer to not have side-effects inside `assert!`.